### PR TITLE
DD-951: add table input_dataset

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes DD-951: add table input_dataset

# Description of changes

# How to test

* [x] run load-from-vault and load-from-fedora twice each

# Related PRs
(Add links)
*

# Notify
@DANS-KNAW/dataversedans

### CODE REVIEW

File: src/main/java/nl/knaw/dans/migration/db/InputDatasetDAO.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  public void create(InputDataset input) {
+    log.trace(input.toString());
+    currentSession().save(input);
+  }
+
+  public void deleteByDoi(String doi, String source) {
+    log.trace("deleting InputDataset {} {}", doi, source);
+    int r = currentSession()
+        .createQuery("DELETE FROM InputDataset WHERE doi = :doi AND source = :source")
```
Comment (jo-pol) [inline src/main/java/nl/knaw/dans/migration/db/InputDatasetDAO.java]: Can the check on `source` cause trouble for load-from-vault when you rerun with a "fixed" store-name? I added the check suspecting datasets might be in fedora as well as in the bag-store.

File: src/main/java/nl/knaw/dans/migration/core/ExpectedLoader.java [hunk 2]
```diff
@@ -30,21 +31,24 @@ public class ExpectedLoader {
...
     this.licensesUrlToName = Mapping.load(new File(configDir + "/licenses.csv"),"url","name");
   }
 
-  public void deleteByDoi(String doi, Mode mode) {
+  public void deleteByDoi(String doi, Mode mode, String source) {
     if (mode.doFiles())
       expectedFileDAO.deleteByDoi(doi);
     if (mode.doDatasets())
       expectedDatasetDAO.deleteByDoi(doi);
+    inputDatasetDAO.deleteByDoi(doi, source);
```
Comment (jo-pol) [inline src/main/java/nl/knaw/dans/migration/core/ExpectedLoader.java]: Don't remember why at some point in time I wanted this last line only for load-from-vault.

File: src/main/java/nl/knaw/dans/migration/db/InputDatasetDAO.java (hunk 1 above)
Comment (lindareijnhoudt) [reply src/main/java/nl/knaw/dans/migration/db/InputDatasetDAO.java]: that's a sensible check, considering the landingpage-only pages we are adding now are taken from Fedora, even from Vault datasets

Comment (lindareijnhoudt) [reply src/main/java/nl/knaw/dans/migration/db/InputDatasetDAO.java]: actually, i would also like to add the batch to this check. So only the current file is reloaded

Comment (lindareijnhoudt) [reply src/main/java/nl/knaw/dans/migration/db/InputDatasetDAO.java]: actually, i would also like to add the batch to this check. So only the current file is reloaded

File: src/main/java/nl/knaw/dans/migration/core/EasyFileLoader.java [hunk 3]
```diff
@@ -51,29 +53,35 @@ public class EasyFileLoader extends ExpectedLoader {
...
-        deleteByDoi(fedoraToBagCsv.getDoi(), mode);
+        deleteByDoi(fedoraToBagCsv.getDoi(), mode, batch, "fedora");
       }
     }
   }
 
   @UnitOfWork("hibernate")
-  public void loadFromCsv(FedoraToBagCsv csv, Mode mode) {
+  public void loadFromCsv(FedoraToBagCsv csv, Mode mode, File csvFile) {
+    inputDatasetDAO.create(new InputDataset(csv,csvFile));
```
Comment (jo-pol) [inline src/main/java/nl/knaw/dans/migration/core/EasyFileLoader.java:76]: Full path or just the file name?

**Pull request merged**